client/lib/url-search/use-url-search.ts

```diff
@@ -0,0 +1,79 @@
+import { useState } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+import debugFactory from 'debug';
+import page from 'page';
+
+const debug = debugFactory( 'calypso:url-search' );
+
+/**
+ * Function for constructing the url to page to. Here are some examples:
+ * 1. { uri: 'google.com', search:'hello' } --> 'google.com?s=hello'
+ * 2. {
+ *     uri: 'wordpress.com/read/search?q=reader+is+awesome',
+ *     search: 'reader is super awesome'
+ *     queryKey: 'q',
+ *    } --> 'wordpress.com/read/search?q=reader+is+super+awesome'
+ *
+ * @param {object} options the options object
+ * @param {string} options.uri the base uri to modify and add a query to
+ * @param {string} options.search the search term
+ * @param {string} [options.queryKey = s] the key to place in the url.  defaults to s
+ * @returns {string} The built search url
+ */
+export const buildSearchUrl = ( {
+	uri,
+	search,
+	queryKey = 's',
+}: {
+	uri: string;
+	search?: string;
+	queryKey?: string;
+} ) => {
+	const query = addQueryArgs( uri, { [ queryKey ]: search } );
+
+	if ( ! search ) {
+		const queryStringIndex = uri.indexOf( '?' );
```

Comment:
Theoretically, a question mark could exist in the hash fragment of the URL with no query arguments:

![](https://cldup.com/-kbg2KYJeW.png)

Should we tighten this up a bit?

Reply:
Yeah, good point. Fortunately, I moved away from this implementation :)